components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/signup/UserSelfRegistrationManager.java

```diff
@@ -794,6 +842,29 @@ private UserStoreManager getUserStoreManager(User user) throws IdentityRecoveryE
         }
     }
 
+    /**
+     * Validate context username and tenant with the stored user's username and tenant domain..
+     *
+     * @param user User
+     * @throws IdentityRecoveryException Invalid Username/Tenant.
+     */
+    private void validateUser(User user) throws IdentityRecoveryException {
+
+        String contextUsername = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        String username = user.getUserName();
+        if (!StringUtils.equalsIgnoreCase(contextUsername, username)) {
+            throw Utils.handleClientException(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_USER,
+                    contextUsername);
+        }
+
+        String contextTenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        String userTenantDomain = user.getTenantDomain();
+        if (!StringUtils.equalsIgnoreCase(contextTenantDomain, userTenantDomain)) {
+            throw Utils.handleClientException(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_TENANT,
+                    contextTenantDomain);
+        }
```

Comment:
There is a code duplication here. Let's directly call this method: https://github.com/wso2-extensions/identity-governance/pull/414/files#diff-924c79caec33e8728c378c88f876c1d6R874

Reply:
Addressed in https://github.com/wso2-extensions/identity-governance/pull/414/commits/b5f0c45e2ce7cb40a24f7a051acc52455755cce4